protobufs/livekit_models.proto

```diff
@@ -447,6 +447,8 @@ enum DisconnectReason {
   USER_UNAVAILABLE = 11;
   // SIP callee rejected the call (busy)
   USER_REJECTED = 12;
+  // SIP protocol failure or unexpected response
+  USER_FAILURE = 13;
```

Comment:
do you have some examples of how this would fail? `USER_FAILURE` seems too generic.. and it seems like it's not an explicit user action

Reply:
A typical failure would be the SIP server we are trying to dial rejecting our request (e.g. wrong password, wrong number, etc).

Agree, the name is not the best one. I considered `EXTERNAL_FAILURE`, `INTEGRATION_FAILURE` or `CONNECTOR_FAILURE` to indicate that it's not an issue with LiveKit, but with the external service.

Reply:
`SIP_TRUNK_FAILURE` ?

Reply:
That would work, but we specifically omitted `SIP_` prefix in other cases. Maybe we could also use a name that could be reused for other integrations?

Reply:
Changed to `SIP_TRUNK_FAILURE`.